gcovr/workers.py

```diff
@@ -95,7 +113,8 @@ def add(self, work, *args, **kwargs):
         Add in a method and the arguments to be used
         when running it
         """
-        self.q.put((work, args, kwargs))
+        if not work or not self.exceptions:
+            self.q.put((work, args, kwargs))
```

Comment:
This methods seems to do: add a work item to the queue unless there already was an exception. Always add the end sentinels. The check is not needed for correctness, but is only used as an optimization to avoid enqueueing items that would be drained immediately. Is that correct?

It might be a lot easier to understand if the end sentinels are added through a different method, and only keep the `if not exceptions: put(…)` check.

Reply:
That is correct. I can make this change if you would like? I agree it would make the code neater to have an `add_sentinel` method.

Reply:
Your call. Perhaps just add a comment. Thankyou for confirming my understanding.

Reply:
I made an add_sentinel method - it fitted well with the refactor to ensure draining and adding work couldn't conflict.